### PR TITLE
 networkd: route add support to configure fastopen_no_cookie 

### PR DESCRIPTION
**Pull request by ssahani**

### CODE REVIEW

File: src/network/networkd-route.c [hunk 1]
```diff
@@ -644,6 +645,12 @@ int route_configure(
                         return log_link_error_errno(link, r, "Could not append RTAX_QUICKACK attribute: %m");
         }
 
+        if (route->fast_open_no_cookie != -1) {
```
Comment (yuwata) [inline src/network/networkd-route.c]: We usually use `>= 0` for tristate.

File: src/network/networkd-route.c [hunk 2]
```diff
@@ -644,6 +645,12 @@ int route_configure(
...
+                r = sd_netlink_message_append_u32(req, RTAX_FASTOPEN_NO_COOKIE, route->quickack);
```
Comment (yuwata) [inline src/network/networkd-route.c]: `route->fast_open_no_cookie`

File: src/network/networkd-route.c [hunk 3]
```diff
@@ -1197,6 +1204,44 @@ int config_parse_quickack(
...
+                return r;
+
+        k = parse_boolean(rvalue);
+        if (k < 0) {
+                log_syntax(unit, LOG_ERR, filename, line, k,
+                           "Failed to parse TCP fastopen no cookie, ignoring: %s", rvalue);
+                return 0;
+        }
+
+        n->fast_open_no_cookie = !!k;
```
Comment (yuwata) [inline src/network/networkd-route.c]: Not a matter, but we usually do not use `!!`, as it is already checked the nagative case in the above.

**Comment by ssahani** [comment]
Updated thanks for the review @yuwata and @poettering .

**Comment by yuwata** [comment]
LGTM.

**Pull request merged**